### PR TITLE
Fix registry refresh

### PR DESCRIPTION
**Pull request by Bolodya1997**
# Issues
1. `refresh` uses name from client.
2. `startRefresh` overrides initial URL with returned from server.
3. `startRefresh` should fail on error returned from server.
4. Tests wait for incorrect `requestCount`.
# Solution
1. Always override only name and expiration time with returned from server.
2. Same.
3. Stop retrying to refresh on error returned, log a warning message.
4. Fix tests.

### CODE REVIEW

File: pkg/registry/common/refresh/nse_registry_client_test.go [hunk 1]
```diff
@@ -18,171 +18,187 @@ package refresh_test
...
+func TestRefreshNSEClient_ShouldSetExpirationTime_BeforeCallNext(t *testing.T) {
+	defer goleak.VerifyNone(t, goleak.IgnoreCurrent())
 
-func (c *checkExpirationTimeClient) Unregister(ctx context.Context, in *registry.NetworkServiceEndpoint, opts ...grpc.CallOption) (*empty.Empty, error) {
-	return new(empty.Empty), nil
-}
+	client := next.NewNetworkServiceEndpointRegistryClient(
+		new(requestCountClient),
+		refresh.NewNetworkServiceEndpointRegistryClient(refresh.WithDefaultExpiryDuration(time.Hour)),
+		&checkNSEClient{
```
Comment (denis-tingaikin) [inline pkg/registry/common/refresh/nse_registry_client_test.go:75]: I think this should be moved to the same place as it is in `netowrkservice/`

**Review by denis-tingaikin — COMMENTED** [review]
In general looks ok

File: pkg/registry/common/refresh/nse_registry_client.go [hunk 2]
```diff
@@ -45,72 +66,77 @@ func (c *refreshNSEClient) startRefresh(ctx context.Context, client registry.Net
...
-				nse.ExpirationTime.Nanos = int32(t1.Nanosecond())
-				var err error
-				nse, err = client.Register(ctx, nse)
+				nse.ExpirationTime = timestamppb.New(time.Now().Add(delta))
+
+				res, err := client.Register(ctx, nse.Clone())
 				if err != nil {
-					<-time.After(c.retryDelay)
-					continue
+					logEntry.Warnf("failed to update registration: %s", err.Error())
```
[the remark this answers is not in the document]
Comment (Bolodya1997) [reply pkg/registry/common/refresh/nse_registry_client.go]: Fixed.

File: pkg/registry/common/refresh/nse_registry_client.go [hunk 3]
```diff
@@ -45,72 +66,77 @@ func (c *refreshNSEClient) startRefresh(ctx context.Context, client registry.Net
...
-		retryDelay:            time.Second * 5,
-		defaultExpiryDuration: time.Minute * 30,
-		chainContext:          context.Background(),
-	}
-
-	for _, o := range options {
-		o.apply(c)
+	resp, err := next.NetworkServiceEndpointRegistryClient(ctx).Unregister(ctx, nse, opts...)
+	if err != nil {
+		return nil, err
```
[the remark this answers is not in the document]
Comment (Bolodya1997) [reply pkg/registry/common/refresh/nse_registry_client.go]: Fixed.

File: pkg/registry/common/refresh/nse_registry_client.go [hunk 4]
```diff
@@ -45,72 +66,77 @@ func (c *refreshNSEClient) startRefresh(ctx context.Context, client registry.Net
...
 	}
 
-	return c
+	return resp, nil
```
[the remark this answers is not in the document]
Comment (Bolodya1997) [reply pkg/registry/common/refresh/nse_registry_client.go]: Fixed.

File: pkg/registry/common/refresh/nse_registry_client_test.go (hunk 1 above)
Comment (Bolodya1997) [reply pkg/registry/common/refresh/nse_registry_client_test.go]: Yep, moved.

File: pkg/registry/common/refresh/nse_registry_client.go [hunk 5]
```diff
@@ -20,23 +20,45 @@ import (
...
+
+	return c
+}
+
+func (c *refreshNSEClient) startRefresh(
+	ctx context.Context,
+	client registry.NetworkServiceEndpointRegistryClient,
+	nse *registry.NetworkServiceEndpoint,
+) {
+	_, logEntry := logruslogger.New(ctx)
```
Comment (denis-tingaikin) [inline pkg/registry/common/refresh/nse_registry_client.go:59]: Please use logger from passed context

**Review by denis-tingaikin — APPROVED** [review]
Could you fix one comment related to logging?

File: pkg/registry/common/refresh/nse_registry_client.go (hunk 5 above)
Comment (Bolodya1997) [reply pkg/registry/common/refresh/nse_registry_client.go:59]: It is a chain context, there can be no logger:
```go
ctx, cancel := context.WithCancel(c.chainContext)
c.startRefresh(ctx, nextClient, refreshNSE)
```

Comment (denis-tingaikin) [reply pkg/registry/common/refresh/nse_registry_client.go:59]: Note: logger will be placed in cmd applications.

Comment (denis-tingaikin) [reply pkg/registry/common/refresh/nse_registry_client.go:59]: See here https://github.com/networkservicemesh/cmd-nsmgr/blob/master/main.go#L46

Comment (Bolodya1997) [reply pkg/registry/common/refresh/nse_registry_client.go]: Yep, fixed.

Comment (Bolodya1997) [reply pkg/registry/common/refresh/nse_registry_client.go]: Now default chain context is:
```go
logger.WithLog(context.Background())
```

File: pkg/registry/common/refresh/nse_registry_client.go [hunk 6]
```diff
@@ -41,7 +41,7 @@ type refreshNSEClient struct {
 func NewNetworkServiceEndpointRegistryClient(options ...Option) registry.NetworkServiceEndpointRegistryClient {
 	c := &refreshNSEClient{
 		defaultExpiryDuration: time.Minute * 30,
-		chainContext:          context.Background(),
+		chainContext:          logger.WithLog(context.Background()),
```
Comment (denis-tingaikin) [inline pkg/registry/common/refresh/nse_registry_client.go:44]: it is ok for now, but should be refactored.

**Pull request merged**